org.eclipse.dawnsci.remotedataset.client/src/org/eclipse/dawnsci/remotedataset/client/RemoteDataHolder.java

```diff
@@ -0,0 +1,193 @@
+package org.eclipse.dawnsci.remotedataset.client;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.dawnsci.analysis.api.io.IDataHolder;
+import org.eclipse.dawnsci.analysis.api.io.IFileLoader;
+import org.eclipse.dawnsci.analysis.api.persistence.IMarshallerService;
+import org.eclipse.dawnsci.analysis.api.tree.Tree;
+import org.eclipse.dawnsci.analysis.tree.TreeToMapUtils;
+import org.eclipse.dawnsci.remotedataset.XMLMarshallerService;
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.ILazyDataset;
+import org.eclipse.january.dataset.LazyDynamicDataset;
+import org.eclipse.january.io.ILazyLoader;
+import org.eclipse.january.metadata.IMetadata;
+
+public class RemoteDataHolder implements IDataHolder {
+	
+	private String path;
+	private String host;
+	private int port;
+	private URLBuilder urlBuilder;
+	private Tree tree;
+	
+	
+	
+	public RemoteDataHolder(String path, String host, int port) {
+		this.urlBuilder = new URLBuilder(host, port);
+		urlBuilder.setPath(path);
+		try {
+			this.tree = TreeToMapUtils.mapToTree(getTreeMap(),path);
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+		}
+	}
+
+	private Map<String, Object> getTreeMap() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)